### ISSUE TITLE
Access to the path '\\?\C:\buildagent\_work\1\a' is denied.

### ISSUE THREAD

**Issue by jrmcdona** [Microsoft/vsts-agent#1165]
## Have you try trouble shooting doc?
Link to trouble shooting doc: https://www.visualstudio.com/en-us/docs/build/troubleshooting

## Agent version and platform
Version of your agent? 2.102.0/2.100.1...

OS of the machine running the agent? OSX

## VSTS type and version
VisualStudio.com
https://microsoft.visualstudio.com/Xbox.Ambassadors/Xbox.Ambassadors%20Team/_build/index?buildId=5955976&_a=summary

## What's not working?
The build was just running fine and I think the server rebooted becasue the Agent stopped.  I logged in and started the build agent in interactive mode like it was before.  I can see it is running again.    But now I get the error:

```
2017-08-28T00:29:47.3538037Z ##[section]Starting: Build
2017-08-28T00:29:47.3538037Z Current agent version: '2.120.1'
2017-08-28T00:29:48.1506635Z ##[section]Starting: Initialize Job
2017-08-28T00:29:48.1662886Z Prepare build directory.
2017-08-28T00:29:48.3225353Z ##[error]Access to the path '\\?\C:\buildagent\_work\1\a' is denied.
2017-08-28T00:29:48.3225353Z ##[section]Finishing: Initialize Job
2017-08-28T00:29:48.9006508Z ##[section]Finishing: Build
```
Also saw this error:
` Access to the path 'C:\buildagent\_work\1\s\.gitattributes' is denied.`

## Agent and Worker's diag log
[2017-08-28 00:25:36Z INFO Program] Version: 2.120.1
[2017-08-28 00:25:36Z INFO Program] Commit: b509fd7b456a81bef8a3d482a0c4619fa39aab46
[2017-08-28 00:25:36Z INFO Program] Culture: en-US
[2017-08-28 00:25:36Z INFO Program] UI Culture: en-US
[2017-08-28 00:25:36Z INFO VstsAgentWebProxy] No proxy setting found.
[2017-08-28 00:25:36Z INFO Worker] Waiting to receive the job message from the channel.
[2017-08-28 00:25:36Z INFO Worker] Message received.
[2017-08-28 00:25:37Z INFO Worker] Job message:
 {
  "tasks": [
    {
      "inputs": {
        "command": "install",
        "workingDir": "Microsoft.Ambassadors/Microsoft.Ambassadors.Web.Client",
        "verbose": "false",
        "customCommand": "",
        "customRegistry": "useNpmrc",
        "customFeed": "",
        "customEndpoint": "",
        "publishRegistry": "useExternalRegistry",
        "publishFeed": "",
        "publishEndpoint": ""
      },
      "instanceId": "b9b232c8-40bc-4f52-8a39-62cfb93ae1a5",
      "displayName": "npm install",
      "enabled": true,
      "condition": "succeeded()",
      "continueOnError": false,
      "alwaysRun": false,
      "timeoutInMinutes": 0,
      "refName": "Npm1",
      "id": "fe47e961-9fa8-4106-8639-368c022d43ad",
      "name": "Npm",
      "version": "1.0.4"
    },
    {
      "inputs": {
        "command": "custom",
        "workingDir": "Microsoft.Ambassadors/Microsoft.Ambassadors.Web.Client",
        "verbose": "false",
        "customCommand": "run build:dev",
        "customRegistry": "useNpmrc",
        "customFeed": "",
        "customEndpoint": "",
        "publishRegistry": "useExternalRegistry",
        "publishFeed": "",
        "publishEndpoint": ""
      },
      "instanceId": "b69c35b3-5106-4fa8-aa71-3c6f557d9e75",
      "displayName": "npm run build:dev",
      "enabled": true,
      "condition": "succeeded()",
      "continueOnError": false,
      "alwaysRun": false,
      "timeoutInMinutes": 0,
      "refName": "Npm2",
      "id": "fe47e961-9fa8-4106-8639-368c022d43ad",
      "name": "Npm",
      "version": "1.0.4"
    },
    {
      "inputs": {
        "command": "restore",
        "publishWebProjects": "true",
        "projects": "",
        "arguments": "",
        "zipAfterPublish": "true"
      },
      "instanceId": "636ed450-896c-47b4-b141-4fc65a550670",
      "displayName": "dotnet restore",
      "enabled": true,
      "condition": "succeeded()",
      "continueOnError": false,
      "alwaysRun": false,
      "timeoutInMinutes": 0,
      "refName": "DotNetCoreCLI4",
      "id": "5541a522-603c-47ad-91fc-a4b1d163081b",
      "name": "DotNetCoreCLI",
      "version": "1.0.2"
    },
    {
      "inputs": {
        "command": "build",
        "publishWebProjects": "true",
        "projects": "",
        "arguments": "",
        "zipAfterPublish": "true"
      },
      "instanceId": "813e2278-fb62-4b39-adeb-0952cdf243f0",
      "displayName": "dotnet build",
      "enabled": true,
      "condition": "succeeded()",
      "continueOnError": false,
      "alwaysRun": false,
      "timeoutInMinutes": 0,
      "refName": "DotNetCoreCLI5",
      "id": "5541a522-603c-47ad-91fc-a4b1d163081b",
      "name": "DotNetCoreCLI",
      "version": "1.0.2"
    },
    {
      "inputs": {
        "ConnectedServiceName": "f60222e3-6cd9-492f-9ec5-4e753110a0b8",
        "WebAppName": "ambassadors-fe",
        "WebAppKind": "app",
        "DeployToSlotFlag": "false",
        "ResourceGroupName": "AmbassadorsFE",
        "SlotName": "",
        "DockerNamespace": "",
        "DockerRepository": "",
        "DockerImageTag": "",
        "StartupCommand": "",
        "VirtualApplication": "",
        "Package": "Microsoft.Ambassadors/Microsoft.Ambassadors.Web.Client",
        "WebAppUri": "http://ambassadors-fe.azurewebsites.net",
        "ScriptType": "",
        "InlineScript": ":: You can provide your deployment commands here. One command per line.",
        "ScriptPath": "",
        "GenerateWebConfig": "false",
        "WebConfigParameters": "",
        "AppSettings": "",
        "TakeAppOfflineFlag": "false",
        "UseWebDeploy": "false",
        "SetParametersFile": "",
        "RemoveAdditionalFilesFlag": "false",
        "ExcludeFilesFromAppDataFlag": "false",
        "AdditionalArguments": "",
        "RenameFilesFlag": "false",
        "XmlTransformation": "false",
        "XmlVariableSubstitution": "false",
        "JSONFiles": ""
      },
      "instanceId": "574990b8-cf54-4dcd-acf9-ab0dc103ccf6",
      "displayName": "Azure App Service Deploy: ambassadors-fe",
      "enabled": true,
      "condition": "succeeded()",
      "continueOnError": false,
      "alwaysRun": false,
      "timeoutInMinutes": 0,
      "refName": "AzureRmWebAppDeployment3",
      "id": "497d490f-eea7-4f2b-ab94-48d9c1acdcb1",
      "name": "AzureRmWebAppDeployment",
      "version": "3.3.13"
    }
  ],
  "requestId": 6690336,
  "lockToken": "00000000-0000-0000-0000-000000000000",
  "lockedUntil": "0001-01-01T00:00:00",
  "messageType": "JobRequest",
  "plan": {
    "scopeIdentifier": "9ed5bb8e-9199-41e1-acaa-00059429261d",
    "planType": "Build",
    "version": 9,
    "planId": "45fdbb12-4a5d-4d64-a89a-17548fbc74a6",
    "artifactUri": "vstfs:///Build/Build/5955974",
    "artifactLocation": null,
    "definition": {
      "_links": {
        "web": {
          "href": "https://microsoft.visualstudio.com/_permalink/_build/index?collectionId=cb55739e-4afe-46a3-970f-1b49d8ee7564&projectId=9ed5bb8e-9199-41e1-acaa-00059429261d&definitionId=16994"
        },
        "self": {
          "href": "https://microsoft.visualstudio.com/9ed5bb8e-9199-41e1-acaa-00059429261d/_apis/build/Definitions/16994"
        }
      },
      "id": 16994,
      "name": "Xbox Ambassadors Front End"
    },
    "owner": {
      "_links": {
        "web": {
          "href": "https://microsoft.visualstudio.com/_permalink/_build/index?collectionId=cb55739e-4afe-46a3-970f-1b49d8ee7564&projectId=9ed5bb8e-9199-41e1-acaa-00059429261d&buildId=5955974"
        },
        "self": {
          "href": "https://microsoft.visualstudio.com/9ed5bb8e-9199-41e1-acaa-00059429261d/_apis/build/Builds/5955974"
        }
      },
      "id": 5955974,
      "name": "5955974"
    }
  },
  "timeline": {
    "id": "12570d29-6501-41c1-8e9f-2c079c855e03",
    "changeId": 1,
    "location": null
  },
  "jobId": "9dcd510d-dea5-4be8-a1e3-6bc56f9057fd",
  "jobName": "Build",
  "jobRefName": "Build",
  "environment": {
    "endpoints": [
      {
        "data": {
          "subscriptionId": "31e13650-96c4-49e0-a261-8925b76d8d77",
          "subscriptionName": "Ambassadors Production",
          "azureSpnRoleAssignmentId": "ff40e661-d257-471a-b040-9edbadcbcd13",
          "azureSpnPermissions": "[{\"roleAssignmentId\":\"ff40e661-d257-471a-b040-9edbadcbcd13\",\"resourceProvider\":\"Microsoft.RoleAssignment\",\"provisioned\":true}]",
          "spnObjectId": "be355054-e274-4319-8a03-fbebe05dc9b9",
          "appObjectId": "94993bcf-3cd9-4542-a3fc-39f59d60df32",
          "creationMode": "Automatic",
          "environment": "AzureCloud",
          "environmentUrl": "https://management.azure.com/",
          "galleryUrl": "https://gallery.azure.com/",
          "serviceManagementUrl": "https://management.core.windows.net/",
          "resourceManagerUrl": "https://management.azure.com/",
          "activeDirectoryAuthority": "https://login.microsoftonline.com/",
          "environmentAuthorityUrl": "https://login.windows.net/",
          "graphUrl": "https://graph.windows.net/",
          "managementPortalUrl": "https://manage.windowsazure.com/",
          "armManagementPortalUrl": "https://portal.azure.com/",
          "activeDirectoryServiceEndpointResourceId": "https://management.core.windows.net/",
          "sqlDatabaseDnsSuffix": ".database.windows.net",
          "AzureKeyVaultDnsSuffix": "vault.azure.net",
          "AzureKeyVaultServiceEndpointResourceId": "https://vault.azure.net"
        },
        "id": "f60222e3-6cd9-492f-9ec5-4e753110a0b8",
        "name": "f60222e3-6cd9-492f-9ec5-4e753110a0b8",
        "type": "azurerm",
        "url": "https://management.azure.com/",
        "authorization": {
          "parameters": {
            "serviceprincipalid": "********",
            "serviceprincipalkey": ********,
            "tenantid": "********"
          },
          "scheme": "ServicePrincipal"
        },
        "isReady": false
      },
      {
        "data": {
          "repositoryId": "2ded624d-0e89-46fc-9124-cf14de3a034f",
          "rootFolder": null,
          "clean": "true",
          "checkoutSubmodules": "False",
          "onpremtfsgit": "False",
          "fetchDepth": "0",
          "gitLfsSupport": "false",
          "skipSyncSource": "false",
          "cleanOptions": "3",
          "checkoutNestedSubmodules": "false"
        },
        "name": "Xbox.Ambassadors",
        "type": "TfsGit",
        "url": "https://microsoft.visualstudio.com/_git/Xbox.Ambassadors",
        "authorization": {
          "parameters": {
            "AccessToken": ********
          },
          "scheme": "OAuth"
        },
        "isReady": false
      }
    ],
    "mask": [
      {
        "type": "regex",
        "value": ********
      },
      {
        "type": "regex",
        "value": ********
      },
      {
        "type": "regex",
        "value": ********
      },
      {
        "type": "regex",
        "value": ********
      },
      {
        "type": "regex",
        "value": ********
      },
      {
        "type": "regex",
        "value": ********
      }
    ],
    "variables": {
      "system": "build",
      "system.collectionId": "cb55739e-4afe-46a3-970f-1b49d8ee7564",
      "system.teamProject": "Xbox.Ambassadors",
      "system.teamProjectId": "9ed5bb8e-9199-41e1-acaa-00059429261d",
      "system.definitionId": "16994",
      "build.definitionName": "Xbox Ambassadors Front End",
      "build.definitionVersion": "11",
      "build.queuedBy": "Jordan McDonald",
      "build.queuedById": "9403d0d9-09a7-40f9-8532-3492e1306734",
      "build.requestedFor": "Jordan McDonald",
      "build.requestedForId": "9403d0d9-09a7-40f9-8532-3492e1306734",
      "build.requestedForEmail": "v-jormc@microsoft.com",
      "build.sourceVersion": "c798ceb44f84948b914c8d9150f0d142de57dba0",
      "build.sourceBranch": "refs/heads/v2Client",
      "build.sourceBranchName": "v2Client",
      "system.debug": "false",
      "build.reason": "Manual",
      "build.buildId": "5955974",
      "build.buildUri": "vstfs:///Build/Build/5955974",
      "build.buildNumber": "5955974",
      "build.containerId": "6602822",
      "system.isScheduled": "False",
      "system.hosttype": "build",
      "system.culture": "en-US",
      "system.teamFoundationCollectionUri": "https://microsoft.visualstudio.com/",
      "system.taskDefinitionsUri": "https://microsoft.visualstudio.com/",
      "AZURE_HTTP_USER_AGENT": "VSTS_cb55739e-4afe-46a3-970f-1b49d8ee7564_build_16994_5955974",
      "MSDEPLOY_HTTP_USER_AGENT": "VSTS_cb55739e-4afe-46a3-970f-1b49d8ee7564_build_16994_5955974",
      "system.planId": "45fdbb12-4a5d-4d64-a89a-17548fbc74a6",
      "system.jobId": "9dcd510d-dea5-4be8-a1e3-6bc56f9057fd",
      "system.timelineId": "12570d29-6501-41c1-8e9f-2c079c855e03",
      "build.repository.uri": "https://microsoft.visualstudio.com/_git/Xbox.Ambassadors",
      "build.sourceVersionAuthor": "Jordan McDonald",
      "build.sourceVersionMessage": "removing npmrc file for now"
    },
    "systemConnection": {
      "data": {
        "ServerId": "cb55739e-4afe-46a3-970f-1b49d8ee7564",
        "ServerName": "microsoft"
      },
      "name": "SystemVssConnection",
      "url": "https://microsoft.visualstudio.com/",
      "authorization": {
        "parameters": {
          "AccessToken": ********
        },
        "scheme": "OAuth"
      },
      "isReady": false
    }
  }
}
[2017-08-28 00:25:37Z INFO JobRunner] Job ID 9dcd510d-dea5-4be8-a1e3-6bc56f9057fd
[2017-08-28 00:25:37Z INFO ConfigurationStore] currentAssemblyLocation: C:\buildagent\bin\Agent.Worker.dll
[2017-08-28 00:25:37Z INFO ConfigurationStore] binPath: C:\buildagent\bin
[2017-08-28 00:25:37Z INFO ConfigurationStore] RootFolder: C:\buildagent
[2017-08-28 00:25:37Z INFO ConfigurationStore] ConfigFilePath: C:\buildagent\.agent
[2017-08-28 00:25:37Z INFO ConfigurationStore] CredFilePath: C:\buildagent\.credentials
[2017-08-28 00:25:37Z INFO ConfigurationStore] ServiceConfigFilePath: C:\buildagent\.service
[2017-08-28 00:25:37Z INFO ConfigurationStore] AutoLogonSettingsFilePath: C:\buildagent\.autologon
[2017-08-28 00:25:38Z INFO JobRunner] Ensure endpoint url match config url base. https://microsoft.visualstudio.com/_git/Xbox.Ambassadors
[2017-08-28 00:25:38Z INFO JobRunner] Ensure System.TaskDefinitionsUrl match config url base. https://microsoft.visualstudio.com/
[2017-08-28 00:25:38Z INFO JobRunner] Ensure System.TFCollectionUrl match config url base. https://microsoft.visualstudio.com/
[2017-08-28 00:25:38Z INFO JobRunner] Ensure SystemConnection url match config url base. https://microsoft.visualstudio.com/
[2017-08-28 00:25:38Z INFO JobRunner] Ensure System.TFServerUrl match config url base. https://microsoft.visualstudio.com/
[2017-08-28 00:25:38Z INFO JobRunner] Creating job server with URL: https://microsoft.visualstudio.com/
[2017-08-28 00:25:38Z INFO VisualStudioServices] Starting operation Location.GetConnectionData
[2017-08-28 00:25:38Z INFO Worker] Listening for cancel message from the channel.
[2017-08-28 00:25:38Z INFO Worker] Waiting for the job to complete or for a cancel message from the channel.
[2017-08-28 00:25:39Z INFO VisualStudioServices] Finished operation Location.GetConnectionData
[2017-08-28 00:25:39Z INFO JobServerQueue] Start process web console line queue.
[2017-08-28 00:25:39Z INFO JobServerQueue] Start process file upload queue.
[2017-08-28 00:25:39Z INFO JobServerQueue] Start process timeline update queue.
[2017-08-28 00:25:39Z INFO JobRunner] Starting the job execution context.
[2017-08-28 00:25:39Z INFO HostContext] Well known directory 'Bin': 'C:\buildagent\bin'
[2017-08-28 00:25:39Z INFO HostContext] Well known directory 'Root': 'C:\buildagent'
[2017-08-28 00:25:39Z INFO HostContext] Well known directory 'Work': 'C:\buildagent\_work'
[2017-08-28 00:25:39Z INFO JobRunner] Validating directory permissions for: 'C:\buildagent\_work'
[2017-08-28 00:25:39Z INFO HostContext] Well known directory 'Bin': 'C:\buildagent\bin'
[2017-08-28 00:25:39Z INFO HostContext] Well known directory 'Root': 'C:\buildagent'
[2017-08-28 00:25:39Z INFO HostContext] Well known directory 'Work': 'C:\buildagent\_work'
[2017-08-28 00:25:39Z INFO HostContext] Well known directory 'Bin': 'C:\buildagent\bin'
[2017-08-28 00:25:39Z INFO HostContext] Well known directory 'Root': 'C:\buildagent'
[2017-08-28 00:25:39Z INFO HostContext] Well known directory 'Work': 'C:\buildagent\_work'
[2017-08-28 00:25:39Z INFO JobRunner] Creating task server with https://microsoft.visualstudio.com/
[2017-08-28 00:25:39Z INFO VisualStudioServices] Starting operation Location.GetConnectionData
[2017-08-28 00:25:39Z INFO JobServerQueue] Try to append 1 batches web console lines, success rate: 1/1.
[2017-08-28 00:25:39Z INFO VisualStudioServices] Finished operation Location.GetConnectionData
[2017-08-28 00:25:40Z INFO JobRunner] Getting job extension.
[2017-08-28 00:25:40Z INFO ExtensionManager] Getting extensions for interface: 'Microsoft.VisualStudio.Services.Agent.Worker.IJobExtension'
[2017-08-28 00:25:40Z INFO ExtensionManager] Creating instance: Microsoft.VisualStudio.Services.Agent.Worker.Build.BuildJobExtension, Agent.Worker
[2017-08-28 00:25:40Z INFO ExtensionManager] Creating instance: Microsoft.VisualStudio.Services.Agent.Worker.Release.ReleaseJobExtension, Agent.Worker
[2017-08-28 00:25:40Z INFO ExtensionManager] Creating instance: Microsoft.VisualStudio.Services.Agent.Worker.Release.DeploymentJobExtension, Agent.Worker
[2017-08-28 00:25:40Z INFO ExtensionManager] Creating instance: Microsoft.VisualStudio.Services.Agent.Worker.Maintenance.MaintenanceJobExtension, Agent.Worker
[2017-08-28 00:25:40Z INFO JobRunner] Initialize job. Getting all job steps.
[2017-08-28 00:25:40Z INFO BuildJobExtension] Run initial step from extension BuildJobExtension.
[2017-08-28 00:25:40Z INFO ExtensionManager] Getting extensions for interface: 'Microsoft.VisualStudio.Services.Agent.Worker.Build.ISourceProvider'
[2017-08-28 00:25:40Z INFO ExtensionManager] Creating instance: Microsoft.VisualStudio.Services.Agent.Worker.Build.ExternalGitSourceProvider, Agent.Worker
[2017-08-28 00:25:40Z INFO ExtensionManager] Creating instance: Microsoft.VisualStudio.Services.Agent.Worker.Build.GitHubSourceProvider, Agent.Worker
[2017-08-28 00:25:40Z INFO ExtensionManager] Creating instance: Microsoft.VisualStudio.Services.Agent.Worker.Build.BitbucketSourceProvider, Agent.Worker
[2017-08-28 00:25:40Z INFO ExtensionManager] Creating instance: Microsoft.VisualStudio.Services.Agent.Worker.Build.SvnSourceProvider, Agent.Worker
[2017-08-28 00:25:40Z INFO ExtensionManager] Creating instance: Microsoft.VisualStudio.Services.Agent.Worker.Build.TfsGitSourceProvider, Agent.Worker
[2017-08-28 00:25:40Z INFO ExtensionManager] Creating instance: Microsoft.VisualStudio.Services.Agent.Worker.Build.TfsVCSourceProvider, Agent.Worker
[2017-08-28 00:25:40Z INFO BuildDirectoryManager] Checking if build directory exists: 'C:\buildagent\_work\1'
[2017-08-28 00:25:40Z ERR  BuildJobExtension] Caught exception from JobExtension Initialization: System.AggregateException: One or more errors occurred. (One or more errors occurred. (Access to the path 'C:\buildagent\_work\1\s\.gitattributes' is denied.)) (Access to the path 'C:\buildagent\_work\1\s\.gitattributes' is denied.) ---> System.UnauthorizedAccessException: Access to the path 'C:\buildagent\_work\1\s\.gitattributes' is denied.
   at System.IO.Win32FileSystem.DeleteFile(String fullPath)
   at Microsoft.VisualStudio.Services.Agent.Util.IOUtil.<>c__DisplayClass21_1.<DeleteDirectory>b__0(FileSystemInfo item)
   at System.Linq.Parallel.ForAllOperator`1.ForAllEnumerator`1.MoveNext(TInput& currentElement, Int32& currentKey)
   at System.Linq.Parallel.ForAllSpoolingTask`2.SpoolingWork()
   at System.Linq.Parallel.SpoolingTaskBase.Work()
   at System.Linq.Parallel.QueryTask.BaseWork(Object unused)
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Linq.Parallel.QueryTaskGroupState.QueryEnd(Boolean userInitiatedDispose)
   at System.Linq.Parallel.MergeExecutor`1.Execute[TKey](PartitionedStream`2 partitions, Boolean ignoreOutput, ParallelMergeOptions options, TaskScheduler taskScheduler, Boolean isOrdered, CancellationState cancellationState, Int32 queryId)
   at System.Linq.Parallel.PartitionedStreamMerger`1.Receive[TKey](PartitionedStream`2 partitionedStream)
   at System.Linq.Parallel.ForAllOperator`1.WrapPartitionedStream[TKey](PartitionedStream`2 inputStream, IPartitionedStreamRecipient`1 recipient, Boolean preferStriping, QuerySettings settings)
   at System.Linq.Parallel.UnaryQueryOperator`2.UnaryQueryOperatorResults.ChildResultsRecipient.Receive[TKey](PartitionedStream`2 inputStream)
   at System.Linq.Parallel.ScanQueryOperator`1.ScanEnumerableQueryOperatorResults.GivePartitionedStream(IPartitionedStreamRecipient`1 recipient)
   at System.Linq.Parallel.UnaryQueryOperator`2.UnaryQueryOperatorResults.GivePartitionedStream(IPartitionedStreamRecipient`1 recipient)
   at System.Linq.Parallel.QueryOperator`1.GetOpenedEnumerator(Nullable`1 mergeOptions, Boolean suppressOrder, Boolean forEffect, QuerySettings querySettings)
   at System.Linq.Parallel.ForAllOperator`1.RunSynchronously()
   at Microsoft.VisualStudio.Services.Agent.Util.IOUtil.DeleteDirectory(String path, Boolean contentsOnly, Boolean continueOnContentDeleteError, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Agent.Worker.Build.BuildDirectoryManager.DeleteDirectory(IExecutionContext executionContext, String description, String path)
   at Microsoft.VisualStudio.Services.Agent.Worker.Build.BuildDirectoryManager.CreateDirectory(IExecutionContext executionContext, String description, String path, Boolean deleteExisting)
   at Microsoft.VisualStudio.Services.Agent.Worker.Build.BuildDirectoryManager.PrepareDirectory(IExecutionContext executionContext, ServiceEndpoint endpoint, ISourceProvider sourceProvider)
   at Microsoft.VisualStudio.Services.Agent.Worker.Build.BuildJobExtension.InitializeJobExtension(IExecutionContext executionContext)
   at Microsoft.VisualStudio.Services.Agent.Worker.JobExtension.<InitializeJob>d__9.MoveNext()
---> (Inner Exception #0) System.UnauthorizedAccessException: Access to the path 'C:\buildagent\_work\1\s\.gitattributes' is denied.
   at System.IO.Win32FileSystem.DeleteFile(String fullPath)
   at Microsoft.VisualStudio.Services.Agent.Util.IOUtil.<>c__DisplayClass21_1.<DeleteDirectory>b__0(FileSystemInfo item)
   at System.Linq.Parallel.ForAllOperator`1.ForAllEnumerator`1.MoveNext(TInput& currentElement, Int32& currentKey)
   at System.Linq.Parallel.ForAllSpoolingTask`2.SpoolingWork()
   at System.Linq.Parallel.SpoolingTaskBase.Work()
   at System.Linq.Parallel.QueryTask.BaseWork(Object unused)
   at System.Threading.Tasks.Task.Execute()<---

[2017-08-28 00:25:40Z ERR  JobRunner] Job initialize failed.
[2017-08-28 00:25:40Z ERR  JobRunner] Caught exception from InitializeJob: System.AggregateException: One or more errors occurred. (One or more errors occurred. (Access to the path 'C:\buildagent\_work\1\s\.gitattributes' is denied.)) (Access to the path 'C:\buildagent\_work\1\s\.gitattributes' is denied.) ---> System.UnauthorizedAccessException: Access to the path 'C:\buildagent\_work\1\s\.gitattributes' is denied.
   at System.IO.Win32FileSystem.DeleteFile(String fullPath)
   at Microsoft.VisualStudio.Services.Agent.Util.IOUtil.<>c__DisplayClass21_1.<DeleteDirectory>b__0(FileSystemInfo item)
   at System.Linq.Parallel.ForAllOperator`1.ForAllEnumerator`1.MoveNext(TInput& currentElement, Int32& currentKey)
   at System.Linq.Parallel.ForAllSpoolingTask`2.SpoolingWork()
   at System.Linq.Parallel.SpoolingTaskBase.Work()
   at System.Linq.Parallel.QueryTask.BaseWork(Object unused)
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Linq.Parallel.QueryTaskGroupState.QueryEnd(Boolean userInitiatedDispose)
   at System.Linq.Parallel.MergeExecutor`1.Execute[TKey](PartitionedStream`2 partitions, Boolean ignoreOutput, ParallelMergeOptions options, TaskScheduler taskScheduler, Boolean isOrdered, CancellationState cancellationState, Int32 queryId)
   at System.Linq.Parallel.PartitionedStreamMerger`1.Receive[TKey](PartitionedStream`2 partitionedStream)
   at System.Linq.Parallel.ForAllOperator`1.WrapPartitionedStream[TKey](PartitionedStream`2 inputStream, IPartitionedStreamRecipient`1 recipient, Boolean preferStriping, QuerySettings settings)
   at System.Linq.Parallel.UnaryQueryOperator`2.UnaryQueryOperatorResults.ChildResultsRecipient.Receive[TKey](PartitionedStream`2 inputStream)
   at System.Linq.Parallel.ScanQueryOperator`1.ScanEnumerableQueryOperatorResults.GivePartitionedStream(IPartitionedStreamRecipient`1 recipient)
   at System.Linq.Parallel.UnaryQueryOperator`2.UnaryQueryOperatorResults.GivePartitionedStream(IPartitionedStreamRecipient`1 recipient)
   at System.Linq.Parallel.QueryOperator`1.GetOpenedEnumerator(Nullable`1 mergeOptions, Boolean suppressOrder, Boolean forEffect, QuerySettings querySettings)
   at System.Linq.Parallel.ForAllOperator`1.RunSynchronously()
   at Microsoft.VisualStudio.Services.Agent.Util.IOUtil.DeleteDirectory(String path, Boolean contentsOnly, Boolean continueOnContentDeleteError, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Agent.Worker.Build.BuildDirectoryManager.DeleteDirectory(IExecutionContext executionContext, String description, String path)
   at Microsoft.VisualStudio.Services.Agent.Worker.Build.BuildDirectoryManager.CreateDirectory(IExecutionContext executionContext, String description, String path, Boolean deleteExisting)
   at Microsoft.VisualStudio.Services.Agent.Worker.Build.BuildDirectoryManager.PrepareDirectory(IExecutionContext executionContext, ServiceEndpoint endpoint, ISourceProvider sourceProvider)
   at Microsoft.VisualStudio.Services.Agent.Worker.Build.BuildJobExtension.InitializeJobExtension(IExecutionContext executionContext)
   at Microsoft.VisualStudio.Services.Agent.Worker.JobExtension.<InitializeJob>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Worker.JobRunner.<RunAsync>d__2.MoveNext()
---> (Inner Exception #0) System.UnauthorizedAccessException: Access to the path 'C:\buildagent\_work\1\s\.gitattributes' is denied.
   at System.IO.Win32FileSystem.DeleteFile(String fullPath)
   at Microsoft.VisualStudio.Services.Agent.Util.IOUtil.<>c__DisplayClass21_1.<DeleteDirectory>b__0(FileSystemInfo item)
   at System.Linq.Parallel.ForAllOperator`1.ForAllEnumerator`1.MoveNext(TInput& currentElement, Int32& currentKey)
   at System.Linq.Parallel.ForAllSpoolingTask`2.SpoolingWork()
   at System.Linq.Parallel.SpoolingTaskBase.Work()
   at System.Linq.Parallel.QueryTask.BaseWork(Object unused)
   at System.Threading.Tasks.Task.Execute()<---

[2017-08-28 00:25:40Z INFO JobServerQueue] Try to append 1 batches web console lines, success rate: 1/1.
[2017-08-28 00:25:40Z INFO JobRunner] Shutting down the job server queue.
[2017-08-28 00:25:40Z INFO JobServerQueue] Fire signal to shutdown all queues.
[2017-08-28 00:25:41Z INFO JobServerQueue] All queue process task stopped.
[2017-08-28 00:25:41Z INFO JobServerQueue] Try to append 1 batches web console lines, success rate: 1/1.
[2017-08-28 00:25:41Z INFO JobServerQueue] Web console line queue drained.
[2017-08-28 00:25:41Z INFO JobServerQueue] Try to upload 2 log files or attachments, success rate: 2/2.
[2017-08-28 00:25:41Z INFO JobServerQueue] File upload queue drained.
[2017-08-28 00:25:41Z INFO JobServerQueue] Timeline update queue drained.
[2017-08-28 00:25:41Z INFO JobServerQueue] All queue process tasks have been stopped, and all queues are drained.
[2017-08-28 00:25:41Z INFO JobRunner] Raising job completed event.
[2017-08-28 00:25:41Z INFO Worker] Job completed.

![image](https://user-images.githubusercontent.com/4327773/29755690-32b00844-8b61-11e7-8da9-c3c66083f3d5.png)

**Comment by TingluoHuang** [comment]
@jrmcdona the error means you don't have permission to access those directory. did you login as the same user before? is there anything change in OS level? Can you find the agent log indicate why it stop?

**Comment by jrmcdona** [comment]
@TingluoHuang  I didn't change anything on the VM.   I logged into it to restart the Agent because I saw the server rebooted.  I am not sure exactly why the server rebooted at this point in time.

The only thing that changed on my build definition was I changed the setting to "Clean" with the clean options to clear source and output becasue my build was deploying same code over and over.

I guess I could just try to rebuild the agent?
